common/src/escrow_accounts.rs

```diff
@@ -99,27 +101,40 @@ type BigInt = String;
 )]
 pub struct EscrowAccountQuery;
 
-pub fn escrow_accounts(
+pub async fn escrow_accounts(
     escrow_subgraph: &'static SubgraphClient,
     indexer_address: Address,
     interval: Duration,
     reject_thawing_signers: bool,
-) -> Eventual<EscrowAccounts> {
-    timer(interval).map_with_retry(
-        move |_| async move {
-            get_escrow_accounts(escrow_subgraph, indexer_address, reject_thawing_signers)
-                .await
-                .map_err(|e| e.to_string())
-        },
-        move |err: String| {
-            error!(
-                "Failed to fetch escrow accounts for indexer {:?}: {}",
-                indexer_address, err
-            );
-
-            sleep(interval.div_f32(2.0))
-        },
-    )
+) -> Receiver<EscrowAccounts> {
+    let initial_accounts =
+        get_escrow_accounts(escrow_subgraph, indexer_address, reject_thawing_signers)
+            .await
+            .expect("Failed to create escrow_accounts channel");
+    let (tx, rx) = watch::channel(initial_accounts);
+    tokio::spawn(async move {
+        let mut time_interval = time::interval(interval);
+        time_interval.set_missed_tick_behavior(time::MissedTickBehavior::Skip);
+        loop {
+            time_interval.tick().await;
+            let result =
+                get_escrow_accounts(escrow_subgraph, indexer_address, reject_thawing_signers).await;
+            match result {
+                Ok(accounts) => tx
+                    .send(accounts)
+                    .expect("Failed to update escrow_accounts channel"),
+                Err(err) => {
+                    error!(
+                        "Failed to fetch escrow accounts for indexer {:?}: {}",
+                        indexer_address, err
+                    );
+                    // Sleep for a bit before we retry
+                    sleep(interval.div_f32(2.0)).await;
+                }
+            }
+        }
+    });
+    rx
```

Comment:
Could you use our newly created function in `watcher.rs`?

Reply:
I was thinking of complete migration to `watcher` in a separate PR, since there are many places we could use that, however I'll update this now.